Switch icon to SVG for Calypso compatibility

mikeyarce opened the pull request:
Currently we don't set an icon in add_menu_page but instead use jQuery to display the icon.  This chnage will make our icon compatible with Calypso so that it will also display the proper icon.

### Changes proposed in this Pull Request

* Switches our icon from a font set with jQuery to just using an SVG icon with `add_menu_page`.

### Testing instructions

* Use this patch
* Go to your dashboard
* Make sure the icon is still there

**To test on Calypso:** 

* Go to a WordPress.com Atomic site
* Install this patched version OR install Sensei LMS and patch the file with these changes
* Make sure that the icon looks good on this url: https://wordpress.com/hosting-config/{YOUR ATOMIC SITE URL}

### Screenshot / Video

**Previously:**
<img width="335" alt="Screenshot 2022-11-22 at 3 47 13 PM" src="https://user-images.githubusercontent.com/3220162/203443691-37da5fbf-8509-4576-81cc-c8774803c18a.png">

**Now:** 
<img width="283" alt="Screenshot 2022-11-22 at 3 47 26 PM" src="https://user-images.githubusercontent.com/3220162/203443702-3cf2ddba-d277-4ec1-9e28-8577e1d7aff7.png">

mikeyarce committed:
Switch icon to SVG for Calypso compatibility

Currently we don't set an icon in add_menu_page but instead use jQuery to display the icon.  This chnage will make our icon compatible with Calypso so that it will also display the proper icon.

mikeyarce committed:
Add member variable doc comment

dadish reviewed (dismissed):
Works great! 🔥

yscik committed:
Change menu icon to bonsai

yscik commented:
I grabbed the opportunity here to update this icon to the bonsai:

<img width="466" alt="image" src="https://user-images.githubusercontent.com/176949/203614169-0c4c8a8a-f855-49ef-940e-2b8b9a9223fe.png">

Issue & discussion:
- #6015 
- p1669223970165419-slack-C013QUH20TS

I'll re-approve to not block the PR, but do take a quick look before merging :)

m1r0 commented:
Nice! I love the new bonsai icon!

I've noticed that it looks a bit off on mobile:

![image](https://user-images.githubusercontent.com/1612178/203803836-42c05fc0-3830-4e9b-a641-5c56a97af714.png)

mikeyarce committed:
Fix icon padding for mobile

mikeyarce commented:
Thanks @m1r0!  Fixed that issue, and thanks @yscik for suggesting an icon update 🎉 

Updated mobile view:
<img width="369" alt="Screenshot 2022-11-24 at 11 55 41 AM" src="https://user-images.githubusercontent.com/3220162/203863449-012d187d-dcbe-4b85-94f3-97c8d8ffe413.png">

m1r0 commented:
@mikeyarce, sorry, but it looks off on desktop now. 😄

mikeyarce committed:
Wrap icon padding in media query for only mobile screens

mikeyarce commented:
😂 Thanks @m1r0 !   I wrapped it in a media query so the padding only shows up in the mobile menu now.

m1r0 reviewed (approved):
Looks great! Thank you!
